### DISCUSSION TITLE
Guice 6.0.0 & 7.0.0 release candidates

### DISCUSSION THREAD

**Discussion by sameb** [google/guice#1732] [Announcements]
Hello Everyone,

I'm pleased to announce that Guice has two new release candidates available:

 * **Guice 6.0.0-rc2**: The [release page](https://github.com/google/guice/wiki/Guice600) has more details, but the TL;DR is that this release fixed a number of bugs, closed all open pull requests, and **added `jakarta.inject` support alongside the existing `javax.inject` support**. (The `jakarta.inject` support is almost complete, with the caveat that binder(..).toProvider() methods aren't supported for jakarta providers.)  Again, see the release notes for complete details & explanations of the javax -> jakarta change.  The **servlet & persist extensions remain javax only for Guice 6.0**. Guice 6.0 is intended as an incremental migration aid towards Guice 7.0.

 * **Guice 7.0.0-rc1**: The [release page](https://github.com/google/guice/wiki/Guice700) has more details, but the TL;DR is that this release is identical to 6.0.0, except it **only supports the jakarta.inject namespace**. The **servlet & persist extensions also only support jakarta.servlet & jakarta.persistence** respectively. **All support for javax.inject, javax.servlet, and javax.persistence are removed from Guice 7.0+.**

Please try out these releases and report back on successes or failures. The more feedback I hear, the faster I can convert these to proper releases. Feedback can be left on on this announcement post, or as issues/PRs in this project.

**I would also like to take a moment to honor Bob Lee's memory**. Bob created Guice (along with Kevin Bourrillion) back around 2007. Although Bob went on to do many other exciting things, Guice's longevity is a testament to his intuition & genius. He was also just a really, really nice guy. I know I will sorely miss Bob, and the world is a worse place without his presence.

**The Guice 6.0 & 7.0 releases are dedicated to Bob.**

**Comment by wendigo** [comment]
I've tested 6.0.0-rc1 in https://github.com/trinodb/trino which is using Guice extensively and everything works flawlessly (https://github.com/trinodb/trino/pull/17286). We've encountered only already discussed issue when running with Azul Prime and we are going to try rc2 to see whether it's resolved. Other than that great work and thank you @sameb! This will unblock us on both Jakarta usage and updating target to JDK 21 :)

**Reply by wendigo** [reply]
rc2 works as well

**Reply by sameb** [reply]
@wendigo -- just to confirm -- rc2 fixed the Azul issue?

**Reply by wendigo** [reply]
@sameb no. Didn't have time for that (other things beside Guice were broken). I can give it a spin tomorrow

**Reply by wendigo** [reply]
@sameb I've verified rc2 with Azul Prime 17 and it works (both ARM64 and AMD64 arch) :)

**Comment by NicoMincuzzi** [comment]
I've just tested `7.0.0-rc1` in my application, which uses also **servlet & persist extensions** and everything seems that works properly! Thank you @sameb!

**Comment by sameb** [comment]
Hi Folks -- Just an update: as of https://github.com/google/guice/commit/bee813b7cc15e46695ca1baf5041a00e0a612f91 I'm happy with the state of things and will soon cut the actual releases (with the newer changes added to each cut). Things were delayed because https://github.com/google/guice/commit/4cea39777a7d253032bc9fae52a62c9ad8ecb87b had some lingering impact on our systems, and I wanted to root-cause what was happening before cutting the real releases.  I now believe I've root caused & fixed it.

**Comment by sameb** [comment]
6.0.0 & 7.0.0 are now released, see announcement @ https://github.com/google/guice/discussions/1742.

**Reply by thisdudeiknew** [reply]
Thanks, @sameb!
